File: samples/java/src/test/java/AndroidSimpleTest.java
import io.appium.java_client.android.AndroidDriver;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.testng.Assert;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;
import java.io.File;
import java.net.URL;


public class AndroidSimpleTest {
    private static AndroidDriver<?> driver;
    private static Logger log= LogManager.getLogger("appium-log");

    @BeforeTest
    public void beforeTest() throws Exception{
        log.info("######################### begin test #########################");
        File classpathRoot = new File(System.getProperty("user.dir"));
        File appPath = new File(classpathRoot, "../../apps/test-app.apk");
        DesiredCapabilities cap = new DesiredCapabilities();
        cap.setCapability("platformName","'Android'");
        cap.setCapability("platformVersion","7.0");
        cap.setCapability("deviceName","520381b347dd148b");
        cap.setCapability("noReset",true);
        cap.setCapability("app",appPath.getCanonicalPath());
//        cap.setCapability("appPackage","io.selendroid.testapp");
//        cap.setCapability("appActivity","io.selendroid.testapp.HomeScreenActivity");
        driver = new AndroidDriver<>(new URL("http://127.0.0.1:4723/wd/hub"), cap);
    }

    @AfterTest
    public void afterTest() {
        driver.quit();
    }

    @Test
    public void testSomething() throws Exception{
        driver.findElementByAccessibilityId("buttonStartWebviewCD").click();
        WebElement input_field = driver.findElementById("name_input");
        input_field.clear();
        input_field.sendKeys("test User");
//        driver.wait(3000);
        log.info(driver.getContextHandles());
        driver.findElementByAndroidUIAutomator("new UiSelector().text(\"Send me your name!\")").click();
        Assert.assertTrue(driver.findElementByAndroidUIAutomator("new UiSelector().text(\"This is my way of saying hello\")").isDisplayed());
        driver.findElementById("goBack").click();
    }
}

File: samples/python/android_simple_test.py
# coding=utf-8
import unittest
from appium import webdriver
import time
import os


class AndroidSimpleTest(unittest.TestCase):
    def setUp(self):
        app_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../apps/test-app.apk'))
        driver_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../drivers'))
        # print os.path.abspath("../../apps/test-app.apk")
        desired_caps = {
            'platformName': 'Android',
            'platformVersion': '7.0',
            'deviceName': '520381b347dd148b',
            "app": app_path,
            # 指定Chromedriver地址，
            # "chromedriverExecutableDir": driver_path,
            # "chromedriverExecutable": r"../drivers/chromedriver.exe",
            # 声明中文
            "unicodeKeyboard": 'True',
            # 声明中文，否则不支持中文
            "resetKeyboard": 'True',
            # 执行时不重新安装包
            'noReset': 'True',
            # 'automationName': 'UiAutomator2',
            'appPackage': 'io.selendroid.testapp',
            'appActivity': 'io.selendroid.testapp.HomeScreenActivity'
        }
        self.driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps)

    def tearDown(self):
        self.driver.quit()

    def test_something(self):
        self.driver.find_element_by_accessibility_id("buttonStartWebviewCD").click()
        input_field = self.driver.find_element_by_id('name_input')
        time.sleep(1)
        input_field.clear()
        input_field.send_keys('Appium User')
        self.driver.find_element_by_android_uiautomator("new UiSelector().text(\"Send me your name!\")").click()
        self.driver.implicitly_wait(4)
        self.assertTrue(self.driver.find_element_by_android_uiautomator("new UiSelector().text(\"This is my way of saying hello\")").is_displayed())
        self.driver.find_element_by_id("goBack").click()


if __name__ == '__main__':
    unittest.main()
